Custom sort2

SFJohnson24 opened the pull request:
This PR introduces Custom columns into our editor and the handling for custom path queries in our CosmosDB integration. 
- updated `containsOperation` to support queries with custom paths that may include nested arrays, using `EXISTS` subqueries with JOINs for complex paths
- added `parseCustomPath` to analyze custom paths and generate appropriate SQL expressions with proper handling of array segments (prefixed with @)
- updated `buildSelect` to handle custom paths separately from standard paths
- Updated `buildJoinsAndFilters` to support filtering on both custom paths and standard paths

It appears these properties queried from SQL do not impact patch/post operations in the dataservice as the yaml rule is fed to the SDK layer and parsed along with user/id/etc.

to test:
run editor locally.  There is a PR attached to this zenhub ticket for the readme showing you how to enter custom column logic.  the column and search features work for different nested keys within rule objects.
save/update/delete rules and check dev DB to ensure changes do not impact these functions.

RamilCDISC reviewed (changes requested):
If you run the editor and remove all the columns, then add a standard column back the screen goes blank. On refresh everything resets and editor is working again. This mean the editor does not crash but just the screen goes blank.

<img width="1504" alt="741" src="https://github.com/user-attachments/assets/4fb8ec39-a67b-4e4a-a6c0-edd69939a661" />

SFJohnson24 commented:
@RamilCDISC I added error handling logic to show a blank.  The issue was the explorerlist was trying to render when before the calls for get_rules was finished causing it to crash.  When the get_rules call is returned and promise filled, a useEffect triggers and a rerender of the list occurs.  It correctly renders a blank before that.   
I also added a stop to prevent removal of the last column so that you cannot remove the last column anymore

RamilCDISC reviewed (changes requested):
I observed that if you add more than one custom column with same check, no rules get loaded. For example create 2 custom columns with `@Authorities.@Standards.Name`. The editor then does not loads any rules. Also if you delete any one of these 2 columns both columns get deleted.

It is recommended that a logic is added that either prevents the user from either making the same column more than once or if it allows multiple custom columns with same name, then the rule loading should not be affected

RamilCDISC reviewed (changes requested):
A bug is observed in editor. If you add a custom column, make some changes to the rule and save it. The editor refreshes the rules list and reflects the changes (if any) in the custom column. But if you now add a default column, make some change to the rule again, the editor rules list does not update and changes are not reflected in the custom column.

Steps to reproduce the bug:
1. Go to locally launched editor and add a custom column with @Authorities.Organization.
2. open a rule (eg CG0019), change the organization property to blank from 'CDISC'
3. save the rule. The rule list will now automatically update and reflect the change to organization property in the custom column.
4. Go to the editor again and put CDISC as value of the Organization property.
5. Save the rule again. This time the list is not updated and the changes are not reflected in the custom column.

SFJohnson24 commented:
I dont reproduce this error at all.  I suspect it is because you were not logged in properly when you were testing editor locally.  I messaged you instructions in slack to get your login information for azure to prevent this--I suspect the lack of a creator is causing this error

RamilCDISC reviewed (approved):
The validation for the Issue is complete. The editor now allows users to create their own custom columns based on the properties of the rule. Also the editor now allows deleting of the default columns. The validation was done with following steps:

1. Check by adding custom column to the editor. (the column is added successfully and shows correct values from the rule).
2. Check for sorting with the custom columns (sorting works as expected).
3. Check deleting and addition of default and custom columns. (works as expected).
4. Check by deleting all the columns. (initially a bug caused editor going blank, now fixed).
5. Check with more than one custom column with same property as check. (initially a bug prevented the rules from loading. Now fixed and editor does not allow adding more than one custom column with same property check).
6. Check if the rule execution features preserved. (works as expected)
7. Check when a default column is added when a custom column already exists. (works fine, however on some systems the rule auto refresh function fails)

More testing with different rules was done to ensure execution. All works as expected.
